src/test/java/io/appium/java_client/service/local/ServerBuilderTest.java

```diff
@@ -1,311 +1,301 @@
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import org.apache.commons.validator.routines.InetAddressValidator;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.openqa.selenium.remote.DesiredCapabilities;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.util.Enumeration;
+import java.util.List;
+
+public class ServerBuilderTest {
+
+    /**
+     * It may be impossible to find the path to the instance of appium server due to different circumstance.
+     * So user may use environment variables/system properties to define the full path to the server
+     * appium.js that is supposed to be default.
+     */
+    private static final String PATH_TO_APPIUM_NODE_IN_PROPERTIES = getProperty(APPIUM_PATH);
+
+    private static final String ROOT_TEST_PATH = "src/test/java/io/appium/java_client/";
+
+    /**
+     * This is the path to the stub appium.js file
+     */
+    private static final String PATH_T0_TEST_APPIUM_JS = ROOT_TEST_PATH + "service/local/appium.js";
+
+    private static String testIP;
+    private AppiumDriverLocalService service;
+    private File testLogFile;
+    private OutputStream stream;
+
+    /**
+     * initialization.
+     */
+    @BeforeClass public static void beforeClass() throws Exception {
+        for (Enumeration<NetworkInterface> en = NetworkInterface.getNetworkInterfaces(); en
+            .hasMoreElements(); ) {
+            NetworkInterface intf = en.nextElement();
+            for (Enumeration<InetAddress> enumIpAddr = intf.getInetAddresses(); enumIpAddr
+                .hasMoreElements(); ) {
+                InetAddress inetAddress = enumIpAddr.nextElement();
+                if (!inetAddress.isLoopbackAddress()) {
+                    InetAddressValidator validator = InetAddressValidator.getInstance();
+
+                    testIP = ofNullable(testIP).orElseGet(() -> {
+                        String result = inetAddress.getHostAddress();
+                        if (validator.isValid(result)) {
+                            return result;
+                        }
+                        return null;
+                    });
+                }
+            }
+        }
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        ofNullable(service).ifPresent(AppiumDriverLocalService::stop);
+
+        if (stream != null) {
+            stream.close();
+        }
+
+        ofNullable(testLogFile).ifPresent(savedTestLogFile -> {
+            if (savedTestLogFile.exists()) {
+                savedTestLogFile.delete();
+            }
+        });
+
+        System.clearProperty(APPIUM_PATH);
+        ofNullable(PATH_TO_APPIUM_NODE_IN_PROPERTIES).ifPresent(s -> setProperty(APPIUM_PATH, s));
```

Comment:
the constant cannot be null

Reply:
@mykola-mokhnach  It may be when the property was not defined